core/src/main/java/com/datastax/oss/driver/api/core/tracker/RequestTracker.java

```diff
@@ -36,6 +36,33 @@
  */
 public interface RequestTracker extends AutoCloseable {
 
+  /**
+   * Invoked each time new request is created.
+   *
+   * @param request the request to be executed
+   * @param executionProfile the execution profile of this request
+   * @param requestLogPrefix the dedicated log prefix for this request
+   */
+  default void onRequestCreated(
+      @NonNull Request request,
+      @NonNull DriverExecutionProfile executionProfile,
+      @NonNull String requestLogPrefix) {}
+
+  /**
+   * Invoked each time a new request is created and sent to next node. Due to retry policy, this
+   * method can be triggered multiple times while processing one logical request.
+   *
+   * @param request the request to be executed
+   * @param executionProfile the execution profile of this request
+   * @param node the node which will receive the request
+   * @param requestLogPrefix the dedicated log prefix for this request
+   */
+  default void onRequestCreatedForNode(
+      @NonNull Request request,
+      @NonNull DriverExecutionProfile executionProfile,
+      @NonNull Node node,
+      @NonNull String requestLogPrefix) {}
```

Comment:
One thing that just popped to mind that might be slightly orthogonal to this change but maybe we should reason about it here, which is SpeculativeExecution;

As its possible that a request may be sent to multiple coordinators (whether retried or speculated), it seems that we will definitively have a way track when nodes get sent requests (`onRequestCreatedForNode`) and also when they get responses (`onNodeSuccess`, `onNodeError`).

But what about in cases where there are requests inflight to multiple nodes?  In the case of speculative retry, I'd expect that when we get the first response, `onNodeSuccess` with the `Node` that got the response first and `onSuccess` will called right away.  What about the other requests that are in in flight?  I suppose there is no callback to `RequestTracker` for those?  It appears that currently we just call `cancel()` on the `NodeResponseCallback` for these in the `setFinalResponse`/`setFinalException` methods.

Also notably, I don't believe there is anything on `ExecutionInfo` even indicating the existence of those requests as they did not fail.